src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -44,16 +44,41 @@ public override BoundNode VisitForEachStatement(BoundForEachStatement node)
                     return RewriteMultiDimensionalArrayForEachStatement(node);
                 }
             }
-            else if (nodeExpressionType.SpecialType == SpecialType.System_String)
+            else if (CanRewriteForeachAsFor(node.Syntax, nodeExpressionType, out var indexerGet, out var lengthGetter))
             {
-                return RewriteStringForEachStatement(node);
+                return RewriteForEachStatementAsFor(node, indexerGet, lengthGetter);
             }
             else
             {
                 return RewriteEnumeratorForEachStatement(node);
             }
         }
 
+        private bool CanRewriteForeachAsFor(SyntaxNode forEachSyntax, TypeSymbol nodeExpressionType, out MethodSymbol indexerGet, out MethodSymbol lengthGet)
```

Comment:
Nit: CanRewrite**Foreach**AsFor => CanRewrite**ForEach**AsFor, to match the `ForEach` casing of other methods?